File: 1910. Remove All Occurrences of a Substring/python/main.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        if len(s) < len(part):
            return s
        
        start_p = 0
        end_p = len(part) - 1
        
        while (end_p< len(s)):
            sub_str = s[start_p:end_p+1]
            if sub_str == part:
                s = s[:start_p] + s[end_p+1:]
                start_p = max(0, start_p - len(part));
                end_p = max(len(part) - 1, end_p - len(part));
                
            else:
                start_p+=1
                end_p+=1
                
                
        return s
    
    # Using built-in func
    def removeOccurrences2(self, s: str, part: str) -> str:
        while part in s:
            s = s.replace(part, '', 1)
        return s

File: 1910. Remove All Occurrences of a Substring/js/main.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function (s, part) {
  if (s.length < part.length) {
    return s;
  }
  // Using sliding window
  let startP = 0;
  let endP = part.length - 1;

  //   If the substring occurs, remove and reduce the startP and endP with the value of part.length
  while (endP < s.length) {
    const substr = s.slice(startP, endP + 1);
    if (substr === part) {
      s = s.substring(0, startP) + s.substring(endP + 1, s.length);
      startP = Math.max(0, startP - part.length);
      endP = Math.max(part.length - 1, endP - part.length);
    } else {
      startP++;
      endP++;
    }
  }
  return s;
};

// using stack (another way)

var removeOccurrences2 = function (s, part) {
  const stack = [];
  const partLength = part.length;

  for (let char of s) {
    stack.push(char);
    // Check if the last partLength characters match the part
    if (stack.length >= partLength) {
      const lastPart = stack.slice(-partLength).join("");
      if (lastPart === part) {
        // Remove the last partLength characters
        stack.splice(-partLength);
      }
    }
  }

  return stack.join("");
};

console.log(removeOccurrences2("axxxxyyyyb", "xy"));
